tools/buildgen/generate_projects.sh

```diff
@@ -18,6 +18,12 @@ set -e
 
 export TEST=${TEST:-false}
 
+YAML_OK=$(python3 -c "import yaml; print(yaml.__version__.split('.') >= ['5', '4', '1'])")
+
+if [[ ${YAML_OK} != "True" ]]; then
+  python3 -m pip install --upgrade --ignore-installed PyYAML==5.4.1 --user
```

Comment:
Any reason not to just unconditionally install this version?

Reply:
The original version upgrades PyYAML unconditionally, but people don't like the extra log when they re-generate the project manually.